### PR TITLE
caretPosition: Simplify `getPosition()`

### PR DESCRIPTION
**Pull request by rhansen**
Multiple commits:
* caretPosition: Invert condition in `getPosition()` for readability
* caretPosition: Delete unused var in `getPosition()`
* caretPosition: Delete no-op `Range.detach()` call
* caretPosition: Delete pointless logic in `getPosition()`
* caretPosition: Clarify comment in `getPosition()`

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Much cleaner, tnx :)  Will get this tested and merged
